src/python/pants/backend/jvm/tasks/jvm_run.py

```diff
@@ -80,6 +80,10 @@ def execute(self):
     else:
       binary = target
 
+    # Some targets will not have extra_jvm_options in their payload,
```

Comment:
Does that mean that this needs a default value? If it's `None`, adding it to `self.jvm_options` below will fail.

Reply:
It has a default value, as added here: https://github.com/pantsbuild/pants/pull/6310/files/38768f87d2803b5b7250f8b7a8497e04d369beb1#r213410501

Reply:
If it's guaranteed to be in the payload, then the comment is inaccurate I think?

Reply:
I think I lost some context from when I wrote this, so I'll probably have to double check. However, I think that the problem was that `jvm_run` was being called with non-`jvm_binary` targets. If that was the case, the problem was that reflection could not find the field, and thus `target.payload.extra_jvm_options` raised an exception. However, `get_field_value` returns the default if the value is not there.

This probably makes the bit [here](https://github.com/pantsbuild/pants/pull/6310/files#r213410501) unnecessary, but I would argue for keeping it.